cloudify_cli/utils.py

```diff
@@ -425,7 +426,21 @@ def _handle_errors():
                     blueprint['error_traceback'])
             raise CloudifyCliError(error_msg)
 
-    blueprint = client.blueprints.get(blueprint_id)
+    @retry(stop_max_attempt_number=5, wait_fixed=1000)
+    def _get_blueprint_and_upload_execution_id():
+        bp = client.blueprints.get(blueprint_id)
+        # upload_execution['id'] might not be available at first, hence retry
+        return bp, bp.upload_execution['id']
+
+    try:
+        blueprint, execution_id = _get_blueprint_and_upload_execution_id()
+    except KeyError:
+        raise RuntimeError(
+            'Failed to get upload_blueprint workflow execution for blueprint '
+            '{0}.  That may indicate a problem with blueprint upload.  Verify '
```

Comment:
Can this sometimes indicate a slow manager, or is it always some other problem?

Reply:
I think currently it can, but soon it will not anymore

Reply:
OK, then I'm happy with this.

Reply:
There could me multiple reasons for a failure, the most basic one would be a break in communication between mgmtworker and restservice.